cpp/CMakeLists.txt

```diff
@@ -294,6 +311,48 @@ else()
     set(NCCL_LIBRARIES ${NCCL_PATH}/lib/libnccl.so)
 endif(NOT NCCL_PATH)
 
+# - FAISS
+if(BUILD_STATIC_FAISS)
+  set(FAISS_DIR ${CMAKE_CURRENT_BINARY_DIR}/faiss CACHE STRING
+    "Path to FAISS source directory")
+  ExternalProject_Add(faiss
```

Comment:
Sorry this wasn't mentioned earlier, but we're migrating to `FetchContent` instead of `ExternalProject` where possible, as see in [this PR](https://github.com/rapidsai/cugraph/pull/1367/files#diff-1bba462ab050e89360fd88110a689e85ee037749cea091a1848ab574381d3795R243).

If this is too involved for this PR, can you add a FIXME?